pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/AbstractExecuteBaseAdminCommand.java

```diff
@@ -0,0 +1,105 @@
+package org.apache.pinot.tools.admin.command;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.net.ssl.SSLContext;
+import org.apache.http.Header;
+import org.apache.http.message.BasicHeader;
+import org.apache.pinot.common.utils.ClientSSLContextGenerator;
+import org.apache.pinot.spi.auth.AuthProvider;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.apache.pinot.spi.utils.CommonConstants;
+import org.apache.pinot.tools.Command;
+import picocli.CommandLine;
+
+public abstract class AbstractExecuteBaseAdminCommand extends AbstractBaseAdminCommand implements Command {
+    @CommandLine.Option(names = {"-controllerHost"}, required = false, description = "Host name for controller.")
+    protected String _controllerHost;
+    @CommandLine.Option(names = {"-controllerPort"}, required = false, description = "Port number for controller.")
+    protected String _controllerPort = DEFAULT_CONTROLLER_PORT;
+    @CommandLine.Option(names = {"-controllerProtocol"}, required = false, description = "Protocol for controller.")
+    protected String _controllerProtocol = CommonConstants.HTTP_PROTOCOL;
+    @CommandLine.Option(names = {"-user"}, required = false, description = "Username for basic auth.")
+    protected String _user;
+    @CommandLine.Option(names = {"-password"}, required = false, description = "Password for basic auth.")
+    protected String _password;
+    @CommandLine.Option(names = {"-authToken"}, required = false, description = "Http auth token.")
+    protected String _authToken;
+    @CommandLine.Option(names = {"-authTokenUrl"}, required = false, description = "Http auth token url.")
+    protected String _authTokenUrl;
+    protected AuthProvider _authProvider;
+    @CommandLine.Option(names = {"-exec"}, required = false, description = "Execute the command.")
+    protected boolean _exec;
+    @CommandLine.Option(names = {"-skipControllerCertValidation"}, required = false, description = "Whether to skip"
+        + " controller certification validation.")
+    private boolean _skipControllerCertValidation = false;
+    @CommandLine.Parameters(paramLabel = "-headers", description = "Additional headers to send.")
+    protected String[] _headers;
+
+    final public AbstractExecuteBaseAdminCommand setControllerHost(String controllerHost) {
```

Comment:
We don't usually put `final` for methods, especially for base class in case we need to override them